### ISSUE TITLE
Add ability to pass CLI args to Wyam config for conditional compilation

### ISSUE THREAD

**Issue by miere43** [statiqdev/Statiq#233]
With passing args you can do conditional compilation of your static content, it can look like that, for example:

`wyam.exe --pass:CompileArticles`

config.wyam

``` c#
===

---

bool compileArticles = Engine.Metadata.Contains("CompileArticles");
bool compileRss = Engine.Metadata.Contains("CompileRss");

if (compileArticles)
{
    Pipelines.Add("Articles",
        ReadFiles("articles/*.md"),
        Markdown(),
        WriteFiles(".html")
    );
}

if (compileRss)
{
    if (!compileArticles) {
        Engine.Trace.Warning("Cannot compile rss because no articles is present.");
    } else {
        Pipelines.Add("Articles (RSS)",
            // ...
        );
    }
}
```

**Comment by daveaglick** [comment]
Agreed - this (or something similar) has been on my radar for a while (see #11). In addition to your use case here for CI (which I hadn't thought of), I also see setting arbitrary values on the command line as being useful for recipes and themes because it would be the most logical way to control them externally (I.e., switching features on and off).

**Comment by miere43** [comment]
Worked a bit on it, seems to be everything is working, check it out right there: 
https://github.com/Wyamio/Wyam/compare/develop...miere43:cli?expand=1
https://github.com/miere43/Wyam/tree/cli

``` c#
===
---
Console.WriteLine(EngineMetadata.Count());
foreach (var meta in EngineMetadata) {
    Console.WriteLine("Got meta: " + meta.Key + " : " + meta.Value);
}
```

**Comment by deanebarker** [comment]
I tried to do this the other day.  My plan was this:

```
--var:foo bar
```

Then in the `Engine`, I would have a `Dictionary<string,string>` where `foo=bar`.  However, I got hung up on the command line parsing library.  I must have tried to do that for an hour before I abandoned.

**Comment by daveaglick** [comment]
Thanks again @miere43 - this was a fantastic contribution. I even expanded the concept a little further by renaming `EngineMetadata` to `GlobalMetadata` and making it and `InitialMetadata` share the same `SimpleMetadata` type (which is really just a `Dictionary<string, object>`). I also added it to `IExecutionContext` as a read-only dictionary. That way, arbitrary objects can also be added from the config file as well as the command line and can be accessed from anywhere via the context.

**Comment by daveaglick** [comment]
Resolved via #237
